CMakeLists.txt

```diff
@@ -120,45 +146,49 @@ set(ZLIB_SRCS
     zutil.c
 )
 
-# append "crc_simd.c" and compile with "mpclmul" if supported by compiler
-if(UNIX)
-    check_c_compiler_flag(-mpclmul HAS_PCLMUL)
-    if(HAS_PCLMUL)
-        set(ENABLE_ASSEMBLY "PCLMUL" CACHE STRING "Choose assembly implementation.")
-        set_property(CACHE ENABLE_ASSEMBLY PROPERTY STRINGS  "OFF;PCLMUL")
-
-        if("${ENABLE_ASSEMBLY}" STREQUAL "PCLMUL")
-            #set(ZLIB_ASMS contrib/amd64/crc32-pclmul_asm.S)
-            #set_source_files_properties(${ZLIB_ASMS} PROPERTIES LANGUAGE C COMPILE_FLAGS -DNO_UNDERLINE)
-            #set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mpclmul")
-            set_source_files_properties(crc32_simd.c PROPERTIES COMPILE_FLAGS -mpclmul)
-            list(APPEND ZLIB_SRCS crc32_simd.c)
-            add_definitions(-DHAS_PCLMUL)
-        endif()
+if(UNIX OR MINGW)
+    # append "inffast_chunk.c" and "adler32_simd.c" for ARMv8 CPU
+    if(ARM_CRC)
+        list(APPEND ZLIB_SRCS inffast_chunk.c adler32_simd.c)
+        add_definitions(-DINFLATE_CHUNK_SIMD_NEON)
+        add_definitions(-DINFLATE_CHUNK_READ_64LE)
+        add_definitions(-DADLER32_SIMD_NEON)
+    endif()
+
+    # append "inffast_chunk.c" and compile with "sse2" if supported by compiler
+    if(HAS_SSE2)
```

Comment:
Both SSE2 and SSSE3 checks are redundant, since they are present on all x86-64 CPUs anyway.

Reply:
Okay, then I will add the flags and definitions directly. I added these two checks just recently since I saw they were checked in the `configure`